syntax = "proto3";

package detective;

// The `DetectiveService` provides APIs.

service DetectiveService {
    rpc GetWitnessMergeDecision(GetWitnessMergeRequest) returns (GetWitnessMergeReply);
    }

enum WitnessEventDecision {
    WITNESS_DECISION_CANT_DECIDE = 0;
    WITNESS_DECISION_MERGE_ALL_POSSIBLE = 1;
    WITNESS_DECISION_PARTIAL_MERGE_POSSIBLE = 2;
    WITNESS_DECISION_NO_MERGE_POSSIBLE = 3;
}

message WitnessEvents {
    repeated string name = 1;
}

message GetWitnessMergeRequest{
    repeated WitnessEvents witness_events = 1;
}

message GetWitnessMergeReply{
    WitnessEventDecision decision = 1;
    repeated WitnessEvents witness_events = 2;
}